doc/sphinx/proofs/writing-proofs/rewriting.rst

```diff
@@ -105,83 +96,47 @@ simply :g:`t=u` dropping the implicit type of :g:`t` and :g:`u`.
-      the form :n:`@term’ = @term`
-
-   .. tacv:: replace @term {? with @term} in @goal_occurrences {? by @tactic}
-             replace -> @term in @goal_occurrences
-             replace <- @term in @goal_occurrences
+   .. exn:: Terms do not have convertible types.
+      :undocumented:
 
-      Acts as before but the replacements take place in the specified clauses
-      (:token:`goal_occurrences`) (see :ref:`performingcomputations`) and not
-      only in the conclusion of the goal. The clause argument must not contain
-      any ``type of`` nor ``value of``.
 
-.. tacn:: subst @ident
-   :name: subst
+.. tacn:: subst {* @ident }
 
-   This tactic applies to a goal that has :n:`@ident` in its context and (at
-   least) one hypothesis, say :g:`H`, of type :n:`@ident = t` or :n:`t = @ident`
-   with :n:`@ident` not occurring in :g:`t`. Then it replaces :n:`@ident` by
-   :g:`t` everywhere in the goal (in the hypotheses and in the conclusion) and
-   clears :n:`@ident` and :g:`H` from the context.
+   For each :n:`@ident`, in order, for which there is a hypothesis in the form
+   :n:`@ident = @term` or :n:`@term = @ident`, replaces :n:`@ident` with :n:`@term`
+   everywhere in the hypotheses and the conclusion and clears :n:`@ident` and the hypothesis
+   from the context.  If there are multiple hypotheses that match the :n:`@ident`,
+   the first one is used.  If no :n:`@ident` is given, replacement is done for all
+   hypotheses in the appropriate form in top to bottom order.
 
-   If :n:`@ident` is a local definition of the form :n:`@ident := t`, it is also
+   If :n:`@ident` is a local definition of the form :n:`@ident := @term`, it is also
    unfolded and cleared.
 
-   If :n:`@ident` is a section variable it is expected to have no
+   If :n:`@ident` is a section variable, it is expected to have no
    indirect occurrences in the goal, i.e. that no global declarations
-   implicitly depending on the section variable must be present in the
+   implicitly depending on the section variable are present in the
    goal.
 
    .. note::
-      + When several hypotheses have the form :n:`@ident = t` or :n:`t = @ident`, the
-        first one is used.
-
-      + If :g:`H` is itself dependent in the goal, it is replaced by the proof of
+      + If the hypothesis is itself dependent in the goal, it is replaced by the proof of
         reflexivity of equality.
 
-   .. tacv:: subst {+ @ident}
-
-      This is equivalent to :n:`subst @ident`:sub:`1`:n:`; ...; subst @ident`:sub:`n`.
-
-   .. tacv:: subst
-
-      This applies :tacn:`subst` repeatedly from top to bottom to all hypotheses of the
-      context for which an equality of the form :n:`@ident = t` or :n:`t = @ident`
-      or :n:`@ident := t` exists, with :n:`@ident` not occurring in
-      ``t`` and :n:`@ident` not a section variable with indirect
-      dependencies in the goal.
-
    .. flag:: Regular Subst Tactic
 
       This flag controls the behavior of :tacn:`subst`. When it is
```

Comment:
There are quite a few `:sub:` worth replacing with a regular `:n:` notation in the paragraph below.